### PR TITLE
Makefile refactoring and extracted zstandard library functions into gpcontrib module

### PR DESCRIPTION
**Pull request by dgkimura**
Extracted zstandard library functions into gpcontrib module.  This is driven by configure option `--with-zstd` which will drive whether to compile zstd module.  In order to help facilitate extracting zstandard, we also moved Greenplum specific modules into gpcontrib modules.

### CODE REVIEW

File: gpcontrib/gpcloud/regress/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-top_builddir = $(shell cd ../../../.. && pwd)
```
Comment (danielgustafsson) [inline gpcontrib/gpcloud/regress/Makefile]: This file lacks the `subdir` variable as well. Not strictly required as it's not used (it should've been, see email to -dev@) but it's good to be consistent.

File: gpcontrib/gpcloud/test/Makefile [hunk 2]
```diff
@@ -27,8 +27,8 @@ DEP_FILES := $(patsubst %.o,%.d,$(TEST_OBJS))
 TEST_SRC = $(TEST_OBJS:.o=.cpp)
 -include $(DEP_FILES)
 
-GMOCK_DIR = ../../googletest/googlemock
-GTEST_DIR = ../../googletest/googletest
+GMOCK_DIR = googletest/googlemock
+GTEST_DIR = googletest/googletest
```
Comment (danielgustafsson) [inline gpcontrib/gpcloud/test/Makefile]: No need for a prefix on these directories?

File: src/Makefile.global.in [hunk 3]
```diff
@@ -221,6 +221,7 @@ PTHREAD_CFLAGS		= @PTHREAD_CFLAGS@
 PTHREAD_LIBS		= @PTHREAD_LIBS@
 
 have_yaml 		= @have_yaml@
+with_zstd = @with_zstd@
```
Comment (danielgustafsson) [inline src/Makefile.global.in]: Indentation

**Comment by hlinnaka** [comment]
I'm on board with the creation of `gpcontrib` and moving the existing contrib modules and extensions there. But what was wrong with having zstandard compressor built-in? The Zlib compressor is also built-in..

**Comment by Eulerizeit** [comment]
@hlinnaka the reasoning for right now is that we want to be able to create compressions as extensions because the current hard coding in the syntax and catalog is weird so we want to address that with another story. 

OTOH, one of the things we have discussed is changing the way we compress query plans and workfiles so that those things can also leverage zstd. If we go that way then this extension approach for zstd clearly won't work and we will need to pull it in like zlib.

**Comment by kmacoskey** [comment]
The generated pipeline is missing here. You should run `concourse/pipelines/gen_pipeline.py -t prod` and commit the newly generated pipeline with the changes to the template. It is possibly that we could move away from that pattern and only commit the template to the repository, but currently that is not the case.

**Pull request merged**